smarthome.go

```diff
@@ -176,7 +217,9 @@ func (s *SmartHome) encodeDeviceForSyncResponse(dev Device) proto.Device {
 }
 
 func (s *SmartHome) handleSyncIntent(agentUserId string) proto.SyncResponse {
-	log.Printf("[%s] SYNC\n", agentUserId)
+	if s.Log != nil {
+		s.Log.Printf("[%s] SYNC REQUEST\n", agentUserId)
+	}
 
 	devices := make([]proto.Device, 0)
```

Comment:
You could move this directly into the struct. 
Don't know which one's better for readability

```
	response := proto.SyncResponse{
		AgentUserId: agentUserId,
		Devices:     make([]proto.Device, 0),
	}
```